File: clean-conflicts.js
#!/usr/bin/env node

const fs = require('fs');
const path = require('path');

// Function to clean merge conflicts
function cleanMergeConflicts(filePath) {
  try {
    if (!fs.existsSync(filePath)) {
      return false;
    }

    let content = fs.readFileSync(filePath, 'utf8');
    let originalContent = content;
    let fixed = false;

    // Check if file has merge conflicts
    if (content.includes('      console.log(`🔧 Cleaning conflicts in: ${filePath}`);

      // Remove merge conflict markers and keep HEAD version (first part)
      content = content.replace(/      
      // Remove any remaining conflict markers
      content = content.replace(/      content = content.replace(/\n?/g, '');
      content = content.replace(/      
      // Clean up extra whitespace and malformed syntax
      content = content.replace(/\n\s*\n\s*\n/g, '\n\n');
      content = content.replace(/^\s+$/gm, '');
      content = content.replace(/;+;+/g, ';');
      content = content.replace(/''+/g, "'");
      content = content.replace(/""+/g, '"');

      fixed = true;
    }

    if (fixed && content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`✅ Cleaned conflicts in: ${filePath}`);
      return true;
    }

    return false;
  } catch (error) {
    console.error(`❌ Error cleaning conflicts in ${filePath}: ${error.message}`);
    return false;
  }
}

// Function to find all files with merge conflicts
function findConflictedFiles() {
  const conflictedFiles = [];

  function scanDirectory(dir) {
    try {
      const items = fs.readdirSync(dir);
      for (const item of items) {
        const fullPath = path.join(dir, item);
        const stat = fs.statSync(fullPath);

        if (stat.isDirectory() && !item.startsWith('.') && item !== 'node_modules') {
          scanDirectory(fullPath);
        } else if (stat.isFile() && (item.endsWith('.ts') || item.endsWith('.tsx') || item.endsWith('.js') || item.endsWith('.jsx') || item.endsWith('.json') || item.endsWith('.md'))) {
          try {
            const content = fs.readFileSync(fullPath, 'utf8');
            if (content.includes('              conflictedFiles.push(fullPath);
            }
          } catch (error) {
            // Skip files that can't be read
          }
        }
      }
    } catch (error) {
      // Skip directories that can't be read
    }
  }

  scanDirectory('.');
  return conflictedFiles;
}

// Main execution
console.log('🚀 Starting merge conflict cleanup...');

const conflictedFiles = findConflictedFiles();
console.log(`Found ${conflictedFiles.length} files with merge conflicts`);

let cleanedCount = 0;
for (const file of conflictedFiles) {
  if (cleanMergeConflicts(file)) {
    cleanedCount++;
  }
}

console.log(`🎉 Cleaned conflicts in ${cleanedCount}/${conflictedFiles.length} files`);

File: clean_conflicts.py
#!/usr/bin/env python3
import os
import re
import glob

def clean_merge_conflicts(file_path):
    """Remove all merge conflict markers and keep the non-HEAD version"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Skip if no conflicts
        if '<<<<<<< HEAD' not in content:
            return False
            
        # Remove all conflict markers and keep content between ======= and >>>>>> 
        # Pattern to match: <<<<<<< HEAD ... ======= (keep this) ... >>>>>> 
        pattern = r'<<<<<<< HEAD.*?=======\s*(.*?)>>>>>>> [^\n]*'
        
        # Replace conflicts with the non-HEAD version
        new_content = re.sub(pattern, r'\1', content, flags=re.DOTALL)
        
        # Clean up any remaining conflict markers
        new_content = re.sub(r'<<<<<<< HEAD.*?=======', '', new_content, flags=re.DOTALL)
        new_content = re.sub(r'>>>>>>> [^\n]*', '', new_content)
        
        # Write back if changed
        if new_content != content:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(new_content)
            return True
            
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False
    
    return False

def main():
    # Find all TypeScript/JavaScript files with conflicts
    patterns = [
        'src/**/*.tsx',
        'src/**/*.ts', 
        'src/**/*.jsx',
        'src/**/*.js'
    ]
    
    files_processed = 0
    files_fixed = 0
    
    for pattern in patterns:
        for file_path in glob.glob(pattern, recursive=True):
            if os.path.isfile(file_path):
                files_processed += 1
                if clean_merge_conflicts(file_path):
                    files_fixed += 1
                    print(f"Fixed: {file_path}")
    
    print(f"Processed {files_processed} files, fixed {files_fixed} files")

if __name__ == "__main__":
    main()